Framework/Core/include/Framework/ASoA.h

```diff
@@ -1389,76 +1389,70 @@ consteval static bool relatedBySortedIndex()
 
 namespace o2::framework
 {
-template <typename T, bool OPT = false, bool SORTED = true>
-struct PresliceBase {
-  constexpr static bool sorted = SORTED;
+
+struct PreslicePolicyBase {
+  const std::string binding;
+  StringPair bindingKey;
+
+  bool isMissing() const;
+  StringPair const& getBindingKey() const;
+};
+
+struct PreslicePolicySorted : public PreslicePolicyBase {
+  void updateSliceInfo(SliceInfoPtr&& si);
+
+  SliceInfoPtr sliceInfo;
+  std::shared_ptr<arrow::Table> getSliceFor(int value, std::shared_ptr<arrow::Table> const& input, uint64_t& offset) const;
+};
+
+struct PreslicePolicyGeneral : public PreslicePolicyBase {
+  void updateSliceInfo(SliceInfoUnsortedPtr&& si);
+
+  SliceInfoUnsortedPtr sliceInfo;
+  gsl::span<const int64_t> getSliceFor(int value) const;
+};
+
+template <typename T, typename Policy, bool OPT = false>
+struct PresliceBase : public Policy {
+  constexpr static bool sorted = std::same_as<Policy, PreslicePolicySorted>;
```

Comment:
Where is this actually needed?

Reply:
Good point, it is not used anywhere now. I will remove it.